tests/bin/close.c

```diff
@@ -0,0 +1,5 @@
+#include "libc.h"
+
+int main(void) {
+    close(STDIN_FILENO);
```

Comment:
To really test this, you should check the return value of `close()` and maybe return `errno` in this case.

Reply:
I'm wondering if the return value from `close` is just implicitly held in `rax` and despite not having a return statement, it is implicitly returned to the system. I think perhaps this should be made explicit.

Reply:
I added an expression to check the return value of `close()`, which should evaluate to false (so zero) if close() returns zero. However, the test now fails and I'm not sure why. I see that close() is [implemented as a proxied call to the host](https://github.com/enarx/enarx-keepldr/blob/master/internal/shim-sgx/src/handler.rs#L543), so there's nothing unusual there. `close()` should return zero on success, so perhaps it's not possible to call `close()` on `STDIN`? Otherwise I'm not quite sure where I went wrong, unfortunately.

I'm not sure how to return `errno`, since the test is expected to return one or zero, but `errno` has multiple error return values.

Reply:
It seems `close()` is not yet implemented in the SEV backend, so this test will fail until we implement it there. It should work with `cargo test --features=backend-sgx`. However, it won't make sense to merge this until `close()` is included in SEV.

Reply:
I implemented close() for the SEV backend, [here](https://github.com/enarx/enarx-keepldr/pull/121).

However, I noticed that if I run `cargo test --features=backend-sgx` against the `close` branch, here, the test for close still fails. So, there may be something else going on and merging #121 may not fix it.